### DISCUSSION TITLE
Display CPU/GPU/RAM Usage/Temp

### DISCUSSION THREAD

**Discussion by satabios** [open-webui/open-webui#7136] [Ideas]
Running models locally can be efficient, but it would be even more useful if we could monitor and display real-time GPU/CPU/RAM usage and temperature metrics directly within the UI.

Currently, I work around this limitation by splitting my screen between btop and either nvidia-smi or nvtop. 

Fortunately, the latest version of btop now includes GPU resource monitoring, which aligns with our needs. Integrating similar functionality into the UI would be a valuable feature for developers like me.

**Comment by flefevre** [comment]
Personaly I have installed Prometheus and Grafana, and I have created a dashboard for GPU monitoring

**Reply by satabios** [reply]
> Personaly I have installed Prometheus and Grafana, and I have created a dashboard for GPU monitoring 

Maybe integrating Prometheus/Grafana could also be an option. Just a small area where the temp/util info would be helpful while using the models.

**Reply by flefevre** [reply]
I could share the dashboard settings without problems

**Reply by j820301** [reply]
@flefevre Hello flefevre, I have completed the installation of Grafana and upgraded openwebui to version 0.6.0, but I haven't seen the relevant configuration method yet. I would like to ask about the steps you took to set it up. Thank you very much for your sharing

**Reply by Lutherwaves** [reply]
Subscribing as I am also interested in the setup

**Comment by reidprichard** [comment]
Agreed. While working out the kinks with this it's not obvious whether A) the model is just loading, B) something crashed so nothing is happening, C) the server is OOM and swapping so it's going to be an indefinite wait, etc. A CPU/memory usage display would be very useful.

**Reply by Wladastic** [reply]
Agreed, I often find myself looking through the openwebui container logs just to find nothing as it passed "something" down and shows 200 but ollama does nothing

**Comment by rgaricano** [comment]
but this is a server side feature, and we can do in differents ways (i use webmin & ssh),
for user side maybe some semaphore lights for network availibility (ws & http api) , server load & general error.
(of course that small semaphore indicators for any ongoing operation as event emmiters signaling it would be perfect for info & debug )

**Comment by pupphelper** [comment]
Being able to see what resources are available VRAM usage especially could be helpful in the end user chosing to use a CPU based model when GPU isn't available or understanding why image generation is slow because VRAM is low.  It would be a nice option to make available to certain groups of users.

**Comment by LaCocoRoco** [comment]
This would be a nice feature. I run comfyui on the same machine and often open the interface just to check the current values.
<img width="668" height="38" alt="image" src="https://github.com/user-attachments/assets/7d0c0bfd-34a9-437e-8c3f-22daa0c7b723" />

**Reply by Zippo2000** [reply]
This is the way I tend to use until now - direct integration to oi would be great

**Reply by Classic298** [reply]
could be doable via filter

**Reply by Zippo2000** [reply]
Can you explain more detail?

**Reply by Classic298** [reply]
build a filter that displays these things above the assistant message

or build an action button that, when clicked, displays these things as a Rich UI

very simple  - you could do exactly what is being discussed here
